### PR TITLE
Skip on any node capability being present

### PR DESCRIPTION
**Pull request by thecoop**
Counterpart to #111268, skip on any capability being present

### CODE REVIEW

File: test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestExecutionContext.java [hunk 1]
```diff
@@ -291,8 +299,46 @@ public Optional<Boolean> clusterHasCapabilities(String method, String path, Stri
             params.put("capabilities", capabilitiesString);
         }
         params.put("error_trace", "false"); // disable error trace
+
+        if (clusterHasFeature(RestNodesCapabilitiesAction.LOCAL_ONLY_CAPABILITIES.id()) == false) {
```
Comment (mosche) [inline test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestExecutionContext.java:303]:
```suggestion
        if (any == false || clusterHasFeature(RestNodesCapabilitiesAction.LOCAL_ONLY_CAPABILITIES.id()) == false) {
```

Isn't this case also what we'd like to use by default if any is `false`? Obviously below works as well, but this one is easier to understand I think.

**Review by mosche — APPROVED** [review]
LGTM 👍

File: test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestExecutionContext.java (hunk 1 above)
Comment (thecoop) [reply test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestExecutionContext.java:303]: I'm on the fence with this one, I think it's helpful to have all the logic used by tests in the same place, so we're not depending on cluster implementation for one and not the other. Could easily be swayed the other way though.

Comment (mosche) [reply test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestExecutionContext.java:303]: I don't have a strong preference to be honest. Consistently doing things a certain way seems beneficial, but I also see value of exercising the default path in tests as much as possible. Leaving it up to you ;)

**Pull request merged**